pygfx/utils/load.py

```diff
@@ -3,74 +3,562 @@
+            raise ValueError(
+                "Loading scenes from URLs is disabled. "
+                "Set remote_ok=True to allow loading from URLs."
+            )
+
+        if not find_spec("httpx"):
+            raise ImportError(
+                "The `httpx` library is required to load meshes from URLs: pip install httpx"
+            )
+        tm_scene = trimesh.load_remote(path)
+    else:
+        tm_scene = trimesh.load(path)
+
+    return scene_from_trimesh(
+        tm_scene,
+        flatten=flatten,
+        meshes=meshes,
+        volumes=volumes,
+        materials=materials,
+        lights=lights,
+        camera=camera,
+    )
+
+
+def meshes_from_trimesh(scene, materials=True, apply_transforms=True):
+    """Converts a trimesh scene into a flat list of pygfx Mesh objects.
+
+    Parameters
+    ----------
+    scene : trimesh.Scene
+        The scene to convert.
+    materials : bool
+        Whether to import materials. If False, a standard material will be created. Default is True.
+    apply_transforms : bool
+        Whether to apply the scene graph transforms directly to the meshes. Default is True.
+
+    Returns
+    -------
+    meshes : list
+        A list of loaded meshes.
+
+    """
     import trimesh  # noqa
 
-    scene = trimesh.load(path)
+    # If this is a single object we can just convert and return it
     if isinstance(scene, trimesh.Trimesh):
         m = gfx.Mesh(
             gfx.geometry_from_trimesh(scene),
-            gfx.MeshPhongMaterial(),
+            gfx.material_from_trimesh(scene),
         )
         return [m]
-
+    # If this is a scene, we need to properly parse it
     elif isinstance(scene, trimesh.Scene):
+        # Scene consists of geometries (which in trimesh contain both
+        # the vertices/faces as well as the mateiral) and a scene graph.
+        # Each node in the graph consists of a geometry and a transform
+        # A geometry can be used by more than one node in the graph.
+        # Here we will convert each node in the graph to a pygfx.Mesh
+        # but in the future it might be better to use pygfx.InstancedMesh.
+
+        # Extract the geometries and materials
+        gfx_geometries, gfx_materials = objects_from_trimesh(scene, materials=materials)
+
+        # Generate a visual for each node
+        meshes = []
         for node_name in scene.graph.nodes_geometry:
             transform, geometry_name = scene.graph[node_name]
-            current = scene.geometry[geometry_name]
-            current.apply_transform(transform)
 
-        return [
-            gfx.Mesh(
-                gfx.geometry_from_trimesh(m),
-                gfx.material_from_trimesh(m.visual.material),
-            )
-            for m in scene.geometry.values()
-        ]
+            mesh = gfx.Mesh(gfx_geometries[geometry_name], gfx_materials[geometry_name])
+
+            if apply_transforms:
+                mesh.local.matrix = transform  # set the node's transform
```

Comment:
Setting the `local.matrix` of an object will not have any effect, unless you load the entire scene as a whole.

If you want to load a Mesh independently, its `local.matrix` property is meaningless, because the `local.matrix` property is only useful when the object is part of a scene or in a coordinate system. In other words, the `local.matrix` only becomes meaningful when the Mesh is added to a parent object, but at that point, the matrix will be recalculated.

Reply:
Are you sure? Only `world.matrix` depends on the parent's transform.

Reply:
> Are you sure? Only `world.matrix` depends on the parent's transform.

Hmm, I'm not entirely sure, and it's possible I may have misunderstood.  but now the transform matrix of the Mesh is not functioning as expected when load from gltf file. If that's the case, the issue might be occurring elsewhere.